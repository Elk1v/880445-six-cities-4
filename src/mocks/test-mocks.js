import moment from "moment";

export const Options = {
  OFFERS_COUNT: 312,
};

export const offers = [
  {
    id: 0,
    title: `Beautiful & luxurious apartment at great location`,
    features: {
      type: `Apartment`,
      bedrooms: 3,
      adults: 4,
    },
    rating: 4.5,
    isPremium: true,
    price: 100,
    images: [
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`
    ],
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    host: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ]
    },

    coords: [52.3909553943508, 4.85309666406198],

    reviews: [{
      id: 0,
      user: {
        name: `Max`,
        avatar: `img/avatar-max.jpg`
      },
      rating: 4.5,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ],
      date: moment().format(`YYYY-MM-DD`),
    }],
  },

  {
    id: 1,
    title: `Beautiful & luxurious apartment at great location`,
    features: {
      type: `Apartment`,
      bedrooms: 3,
      adults: 4,
    },
    rating: 4.0,
    isPremium: false,
    price: 70,
    images: [
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`
    ],
    inside: [
      `Wi-Fi`,
      `Towels`,
      `Coffee machine`,
      `Kitchen`,
      `Dishwasher`,
      `Fridge`,
    ],
    host: {
      name: `Max`,
      avatar: `img/avatar-max.jpg`,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ]
    },

    coords: [52.369553943508, 4.85309666406198],

    reviews: [{
      id: 1,
      user: {
        name: `Max`,
        avatar: `img/avatar-max.jpg`
      },
      rating: 4.5,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ],
      date: moment().format(`YYYY-MM-DD`),
    }],
  },

  {
    id: 2,
    title: `Beautiful & luxurious apartment at great location`,
    features: {
      type: `Apartment`,
      bedrooms: 3,
      adults: 4,
    },
    rating: 4.5,
    isPremium: true,
    price: 100,
    images: [
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`
    ],
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    host: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ]
    },

    coords: [52.3909553943508, 4.929309666406198],

    reviews: [{
      id: 2,
      user: {
        name: `Max`,
        avatar: `img/avatar-max.jpg`
      },
      rating: 4.5,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ],
      date: moment().format(`YYYY-MM-DD`),
    }],
  },

  {
    id: 3,
    title: `Beautiful & luxurious apartment at great location`,
    features: {
      type: `Apartment`,
      bedrooms: 3,
      adults: 4,
    },
    rating: 4.5,
    isPremium: true,
    price: 100,
    images: [
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`
    ],
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    host: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ]
    },

    coords: [52.3809553943508, 4.939309666406198],

    reviews: [{
      id: 3,
      user: {
        name: `Max`,
        avatar: `img/avatar-max.jpg`
      },
      rating: 4.5,
      text: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      ],
      date: moment().format(`YYYY-MM-DD`),
    }],
  },

];

export const offer = {
  id: 0,
  title: `Beautiful & luxurious apartment at great location`,
  features: {
    type: `Apartment`,
    bedrooms: 3,
    adults: 4,
  },
  rating: 4.5,
  isPremium: true,
  price: 100,
  images: [
    `img/studio-01.jpg`,
    `img/apartment-01.jpg`,
    `img/apartment-02.jpg`,
    `img/apartment-03.jpg`,
    `img/studio-01.jpg`,
    `img/apartment-01.jpg`
  ],
  inside: [
    `Wi-Fi`,
    `Washing machine`,
    `Towels`,
    `Heating`,
    `Coffee machine`,
    `Baby seat`,
    `Kitchen`,
    `Dishwasher`,
    `Cabel TV`,
    `Fridge`,
  ],
  host: {
    name: `Angelina`,
    avatar: `img/avatar-angelina.jpg`,
    text: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ]
  },

  coords: [52.3809553943508, 4.939309666406198],

  reviews: {
    id: 0,
    user: {
      name: `Max`,
      avatar: `img/avatar-max.jpg`
    },
    rating: 4.5,
    text: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    date: moment().format(`YYYY-MM-DD`),
  },
};
