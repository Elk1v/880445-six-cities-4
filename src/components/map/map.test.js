import React from "react";
import renderer from "react-test-renderer";
import Map from "Map/map.jsx";
import {offers} from "Mocks/test-mocks";


it(`Map conponent renders correctly`, () => {
  const tree = renderer.create(
      <Map offers={offers}/>,
      {
        createNodeMock: () => document.createElement(`div`)
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
