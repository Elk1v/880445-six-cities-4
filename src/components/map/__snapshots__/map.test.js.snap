// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map conponent renders correctly 1`] = `
<div
  style={
    Object {
      "height": 526,
      "width": 512,
    }
  }
/>
`;
